qiskit_quantuminspire/qi_jobs.py

```diff
@@ -35,15 +59,92 @@ def __init__(
         super().__init__(backend, job_id, **kwargs)
         self._job_ids: List[str] = []
         self._run_input = run_input
+        self.program_name = "Program & Algorithm created by SDK"
 
-    def submit(self) -> None:
+    async def submit(self) -> None:
         """Submit the (batch)job to the quantum inspire backend.
 
         Use compute-api-client to call the cjm endpoints in the correct order, to submit the jobs.
         """
-        for i in range(1, 3):
-            self._job_ids.append(str(i))
-        self.job_id = "999"  # ID of the submitted batch-job
+        options = self.backend().options
+        configuration = config()
+        # call create algorithm
+        async with ApiClient(configuration) as api_client:
+
+            # always create a list
+            circuits: List[QuantumCircuit] = []
+            if isinstance(self._run_input, QuantumCircuit):
+                circuits.append(self._run_input)
+            else:
+                circuits = self._run_input
+
+            project = await self._create_project(api_client, configuration.auth_settings().team_member_id)
+            batch_job = await self._create_batch_job(api_client, backend_type_id=self.backend().id)
+
+            # iterate over the circuits
+            for circuit in circuits:
+                algorithm = await self._create_algorithm(api_client, project)
+                commit = await self._create_commit(api_client, algorithm)
+                file = await self._create_file(api_client, commit, circuit)
+                job: Job = await self._create_job(
+                    api_client, file, batch_job, number_of_shots=options.get("shots", default=1000)
+                )
+                self._job_ids.append(job.job_id())
+
+            await self._enqueue_batch_job(api_client, batch_job)
+            self.job_id = batch_job.id
+
```

Comment:
Many of the create functions below require entire (e.g.) `Commit` or `File` objects as arguments when they only need their ID, this can be simplified.

Reply:
Changed to accept only the int of the id.